### PR TITLE
feat(GaussDBforMySQL): gaussdb mysql instance support auto scaling

### PR DESCRIPTION
**Pull request by houpeng80**
**What this PR does / why we need it**:
  gaussdb mysql instance support auto scaling
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
  gaussdb mysql instance support auto scaling
```

## PR Checklist

* [ ] Tests added/passed.

```
make testacc TEST=./huaweicloud/services/acceptance/gaussdb/ TESTARGS='-run TestAccGaussDBInstance_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/gaussdb/ -v -run TestAccGaussDBInstance_ -timeout 360m -parallel 4
=== RUN   TestAccGaussDBInstance_basic
=== PAUSE TestAccGaussDBInstance_basic
=== RUN   TestAccGaussDBInstance_prePaid
=== PAUSE TestAccGaussDBInstance_prePaid
=== RUN   TestAccGaussDBInstance_updateWithEpsId
=== PAUSE TestAccGaussDBInstance_updateWithEpsId
=== CONT  TestAccGaussDBInstance_basic
=== CONT  TestAccGaussDBInstance_updateWithEpsId
=== CONT  TestAccGaussDBInstance_prePaid
=== CONT  TestAccGaussDBInstance_updateWithEpsId
    acceptance.go:629: The environment variables does not support Migrate Enterprise Project ID for acc tests
--- SKIP: TestAccGaussDBInstance_updateWithEpsId (0.00s)
--- PASS: TestAccGaussDBInstance_prePaid (1424.77s)
--- PASS: TestAccGaussDBInstance_basic (3265.53s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/gaussdb   3265.572s
```

* [ ] Documentation updated.
* [ ] Schema updated.
* [ ] CheckDeleted.

  - **a. During query operation (Read Context)**
    aa. Resource not found
    \>>>>>> Paste the screenshot here <<<<<<

    
  - **b. During delete/disassociate/unbind operation (Delete Context)**
    ba. Resource not found
    \>>>>>> Paste the screenshot here <<<<<<

### CODE REVIEW

File: docs/resources/gaussdb_mysql_instance.md [hunk 1]
```diff
@@ -188,6 +191,56 @@ The `parameters` block supports:
...
+
+* `scaling_strategy` - (Required, List) Specifies the auto-scaling policy.
+  The [scaling_strategy](#scaling_strategy_struct) structure is documented below.
+
+* `monitor_cycle` - (Optional, Int) Specifies the observation period, in seconds. During the entire observation period,
+  if the average CPU usage is greater than or equal to the preset value, a scale-up is triggered. It is mandatory when
+  `status` is set to **ON**. Value options: **300**, **600**, **900** or **1800**.
+
+* `silence_cycle` - (Optional, Int) Specifies the observation period, in seconds. During the entire observation period,
+  if the average CPU usage is greater than or equal to the preset value, a scale-up is triggered. It is mandatory when
```
Comment (saf3dfsa) [inline docs/resources/gaussdb_mysql_instance.md]: wrong description

File: docs/resources/gaussdb_mysql_instance.md [hunk 2]
```diff
@@ -222,6 +280,17 @@ The `nodes` block contains:
 
 * `availability_zone` - Indicates the availability zone where the node resides.
 
+<a name="auto_scaling_struct"></a>
+The `auto_scaling` block supports:
+
+* `id` - Indicates the ID of an auto-scaling policy.
+
+* `min_flavor` - Indicates the minimum specifications..
```
Comment (saf3dfsa) [inline docs/resources/gaussdb_mysql_instance.md]:
```suggestion
* `min_flavor` - Indicates the minimum specifications.
```

File: docs/resources/gaussdb_mysql_instance.md (hunk 1 above)
Comment (houpeng80) [reply docs/resources/gaussdb_mysql_instance.md]: done

File: docs/resources/gaussdb_mysql_instance.md (hunk 2 above)
Comment (houpeng80) [reply docs/resources/gaussdb_mysql_instance.md]: done

File: docs/resources/gaussdb_mysql_instance.md [hunk 3]
```diff
@@ -206,8 +258,13 @@ In addition to all arguments above, the following attributes are exported:
 
 * `updated_at` - Indicates the Update time in the **yyyy-mm-ddThh:mm:ssZ** format.
 
-* `nodes` - Indicates the instance nodes information. Structure is documented below.
+* `nodes` - Indicates the instance nodes information.
+  The [nodes](#nodes_struct) structure is documented below.
+
+* `auto_scaling` - (Optional, List) Indicates the auto-scaling policies.
```
Comment (saf3dfsa) [inline docs/resources/gaussdb_mysql_instance.md]:
```suggestion
* `auto_scaling` - Indicates the auto-scaling policies.
```

**Pull request merged**